lib/kitchen/provisioner/habitat.rb

```diff
@@ -113,104 +107,147 @@ def create_sandbox
       end
 
       def prepare_command
-        wrap_shell_code <<-EOH
-          #{export_hab_bldr_url}
-          #{export_hab_license}
-          #{install_supervisor_command}
-          #{binlink_supervisor_command}
-          #{install_service_package}
+        debug("Prepare command is running")
+        wrap_shell_code <<~PREPARE
           #{remove_previous_user_toml}
           #{copy_user_toml_to_service_directory}
-        EOH
+        PREPARE
       end
 
       def run_command
-        run = <<-RUN
-        #{export_hab_bldr_url}
-        #{export_hab_license}
-        #{clean_up_screen_sessions}
-        #{clean_up_previous_supervisor}
-        echo "Running #{package_ident}."
-        #{run_package_in_background}
-        RUN
-
-        wrap_shell_code run
+        # TODO: This isn't waiting for the service to come up... should we wait? If so, how long?
+
+        # This little bit figures out what package should be loaded
+        if config[:install_latest_artifact] || !config[:artifact_name].nil?
+          target_pkg = get_artifact_name
+          target_ident = "#{config[:package_origin]}/#{config[:package_name]}"
+          # TODO: This is a workaround for windows. The hart file sometimes gets copied to the
+          # %TEMP%\kitchen instead of %TEMP%\kitchen\results.
+          if windows_os?
+            target_pkg = target_pkg.gsub("results/", "") unless File.exist?(target_pkg)
+          end
+        else
+          target_pkg = package_ident
+          target_ident = package_ident
+        end
+
+        if windows_os?
+          wrap_shell_code <<~PWSH
+            if (!($env:Path | Select-String "Habitat")) {
+              $env:Path += ";C:\\ProgramData\\Habitat"
+            }
+            hab pkg install #{target_pkg} --channel #{config[:channel]} --force
+            if (Test-Path -Path "$(hab pkg path #{target_ident})\\hooks\\run") {
+              hab svc load #{target_ident} #{service_options} --force
+              Do {
+                Start-Sleep -Seconds 1
+              } until( hab svc status | out-string -stream | select-string #{target_ident})
```

Comment:
I'm really hesitant to put a retry loop in here without a maximum failure time. Can you update this to retry for 60 seconds, and if the service hasn't started by then fail out with an error message?

Reply:
I added a timeout for both the powershell and bash sections